streams/catalog/src/main/java/com/hortonworks/streamline/streams/catalog/UDF.java

```diff
@@ -112,12 +112,20 @@ public void setType(Type type) {
         this.type = type;
     }
 
+    /**
+     * Canonical class name - see {@link #setClassName}
+     * @return className set when UDF was created
+     */
     public String getClassName() {
         return className;
     }
 
+    /**
+     * Set inner-class names to canonical form if given fqdn form (with '$')
+     * @param className
+     */
     public void setClassName(String className) {
-        this.className = className;
+        this.className = className.replace('$', '.');
```

Comment:
Thanks @arunmahadevan .  I will add those tests, and instead of forcing to canonical form in the setClassName call, I'll force it to fqdn name in ProcessUdf():L313, right before updateTypeInfo().